### DISCUSSION TITLE
Breakpad Crash Handler Not Sending Crash Errors

### DISCUSSION THREAD

**Discussion by ricolaw-clarius** [getsentry/sentry-native#854] [Q&A] [answered]
Hi currently I have set up Sentry for my project along with custom transports via setting sentry_tranport_new() which works properly when creating events with sentry_capture_event(). However it does not seem to send anything when I crash the program.  From what I read in another issue here Crashpad has it's own transport method, is similar for Breakpad? If so is there a way to use our own transport code, or is that for in proc error handling only?

Thanks

**Comment by Swatinem** [comment]
The breakpad handler (and in-proc) will write the event directly to disk in the case of a hard crash. On restart, the event will be loaded from disk and passed to the configured transport.

**Reply by tian-yu-ocado** [reply]
Hi @Swatinem, i am wondering is there a way to send out the crash event before the program completely shuts down, instead of having to wait until the next time the program restarts? That sort of "defeat" the purpose of crash monitoring (you dont know a crash happened until the crash has been recovered)?

**Reply by supervacuus** [reply, marked as answer]
Your question is answered partially here: https://github.com/getsentry/sentry-native/issues/786. The other part is that running an HTTP client inside an already hard-crashed application is rarely a good idea (even if technically possible). 

On Linux specifically we are very limited in what we can do within a signal-handler. There are ways around this, but they haven't been attempted yet, because users can use our crashpad backend in that case (which uses a separate process to send the crash-event "live"). 

Most users gather crash-events over a wide range of deployed target systems in an aggregate fashion where it isn't necessary to be informed of a singular "live" crash event, but rather get an overview of how many of their users or client-configurations are affected by issues in a particular version of their deployables. So yes, with our breakpad- and inproc-backends "monitoring" of a particular deplyoment is not supported yet, although you will be able to retrieve a lot of context that led to the crash after the fact.

**Comment by ricolaw-clarius** [comment]
Is there a method you'd recommend for debugging whether the crash is being saved and whether the event is read again after the program restarts? I am assuming this second step will still use the custom transport.

**Comment by Swatinem** [comment]
> I am assuming this second step will still use the custom transport.

Yes thats true.

You can take a look at the directory that you configured via:

https://github.com/getsentry/sentry-native/blob/0e94e49f442b75da4cf4048e25d577cd581a7b09/include/sentry.h#L960-L990
